server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java

```diff
@@ -675,7 +675,7 @@ public void testNumericLongOrDateSortOptimization() throws Exception {
         searchContext.sort(sortAndFormats);
         searchContext.parsedQuery(new ParsedQuery(new MatchAllDocsQuery()));
         searchContext.setTask(new SearchShardTask(123L, "", "", "", null, Collections.emptyMap()));
-        searchContext.setSize(10);
+        searchContext.setSize(numDocs);
```

Comment:
This part feels wrong, we should instead use `trackTotalHitsUpTo()` or fix `assertSortResults` to no longer fail if the expected hit count is greater than the actual hit count and relation=`gte`? @mayya-sharipova @jimczi opinions?

Reply:
Agreed, I think we should fix `assertSortResults` since the optimization could be disabled (not at the moment) if `trackTotalHitsUpTo` is greater than or equals to the number  of the docs in the index ?

Reply:
I am thinking in fixing it like:

```
  // assert score docs are in order and their number is as expected
    private void assertSortResults(TopDocs topDocs, long numDocs, long trackTotalHitsUpTo, boolean isDoubleSort) {
        assertThat(topDocs.totalHits.value, greaterThanOrEqualTo(trackTotalHitsUpTo));
        assertThat(topDocs.totalHits.value, lessThanOrEqualTo(numDocs));
        long cur1, cur2;
 ......
}
```

@jimczi wdyt?

Reply:
Looks good